crates/storage/trie.rs

```diff
@@ -54,7 +54,7 @@ impl Trie {
     pub fn open(trie_dir: &str, root: H256) -> Result<Self, StoreError> {
```

Comment:
should we change this to init and pass a TrieState as argument? So we can decouple how the trie is stored (for example, in memory doesn't need a trie_dir) with the trie itself

Reply:
I was thinking that with the refactor to DB-trait we would receive a DB-trait object in the trie's init (removing the need for open, create, etc, as we would receive an already-open DB). The cache wouldn't be shared by multiple instances of the trie.

Reply:
Ok